### DISCUSSION TITLE
✅ Hubs v0.0.1 – finops-hub template

### DISCUSSION THREAD

**Discussion by flanakin** [microsoft/finops-toolkit#44] [Releases]
This discussion is focused on v0.0.1, which includes a **finops-hub** template that deploys a storage account and Data Factory pipeline to cleanup duplicate files in storage. This is the most basic release we can have, which will require a bit of manual configuration to setup the export and point it to the hub storage, then setting up Power BI using the out-of-the-box starter reports. This first release will target EA only due to schema inconsistencies across EA, MCA, MOSA, and internal (Microsoft) accounts. We'll address these gaps in a subsequent release but wanted to at least share something for the audience we get the most feedback from (EA).

We're looking for any feedback. To get you started:
1. Does this initial scope for an alpha version make sense?
2. Are we missing anything you'd expect in a first alpha release?
3. Would you deploy the alpha release? If not, what is missing for you to deploy it?
4. Do you have any questions or concerns about deploying FinOps toolkit in general?
5. Thinking about going past v0.0.1, what do you want to see next?

🚀 Release tracker: [Power BI on CM exports](https://github.com/microsoft/cloud-hubs/issues/1)
📋 Milestone tasks: [v0.0.1](https://github.com/microsoft/cloud-hubs/milestone/1)
🌱 ~Working branch: features/hub~

---

### 📥 [Download v0.0.1](/microsoft/cloud-hubs/releases/v0.0.1) &nbsp; &nbsp; &nbsp; 📦 [View template](https://aka.ms/finops/hubs/template) &nbsp; &nbsp; &nbsp; ➕ [Deploy](https://aka.ms/finops/hubs/create) &nbsp; &nbsp; &nbsp; 📗 [Learn more](https://aka.ms/finops/hubs)

**Comment by renanalmeidjr** [comment]
Hi team. We tried working with the Finops tookit and got 4 possible issues, could you please confirm if this is actually a problem, or if we are doing something wrong? 

1) The names of the fields in the parquet files are coming with first letter in lowercase and the mapping in Power Query "Expanded Columns" step is considering the first letter as uppercase. This caused most of columns to be empty.
2) The columns BillingAccountId and BillingProfileId were transformed into "Int" datatype but in our extraction it was GUID (text)
3) Our parquet didn't have the column ResourceName (we tried using this formula to get resourname based on  resourceID but it doesn't work for AKS for example: **_= Table.ReplaceValue(#"Replaced Value",each [ResourceName],each Text.AfterDelimiter([ResourceId], "/", {0, RelativePosition.FromEnd}),Replacer.ReplaceValue,{"ResourceName"})_** )
4) In the Powerquery step "Add TagsString" the Powerquery function **_"{" & [Tags] & "}"_** was not working because it duplicates the JSON braces

**Reply by flanakin** [reply]
Thanks for reaching out! Sorry to hear you're running into issues.
1. Sounds like you're using Microsoft Customer Agreement. The Power BI reports are setup for EA only today. We have this ready for our 0.2 release (October). I'll see about getting it into the 0.1 release as well (next few weeks).
2. @MSBrett any thoughts on this? I'm assuming it's already resolved in 0.2, given the ingestion and normalization changes?
3. This is a gap in Cost Management data, unfortunately. @MSBrett, did you cover this in 0.2?
4. This is another difference between EA and MCA. That said, this should be an easy one to fix by checking the first character for "{". I'll confirm that's also covered in 0.2.

**Reply by flanakin** [reply]
Update on this: The Power BI reports in 0.1 and [0.1.1](https://github.com/microsoft/finops-toolkit/releases/tag/v0.1.1) both support MCA. The parquet fields are still lowercase, since that's coming from Cost Management, but we already have a fix for that queued up with 0.2. This should resolve both issues 1, 3, and 4.
